### PR TITLE
remove language restrictions in tydiqa + add arabic prompts

### PR DESCRIPTION
**Pull request by KhalidAlt**
- [x] Remove the if statement to allow English prompts to work across the Dataset.
- [x] Add Arabic prompts for primary task subset with if statement to include only Arabic set.
- [x] Add Arabic prompts for secondary task subset.

### CODE REVIEW

**Comment by stephenbach** [comment]
Can you make this PR target the eval-hackathon branch? Thanks!

**Comment by VictorSanh** [comment]
@KhalidAlt what is `template.yaml.save` and `template.yaml.save.1`? Can we remove them?

**Comment by KhalidAlt** [comment]
Hi @VictorSanh, I think you can remove them.

**Comment by VictorSanh** [comment]
@KhalidAlt there are still errors in tests for some reason, if you can get to it before please do, I have a few things to sort out this afternoon first

**Comment by stephenbach** [comment]
Sorry, I took so long to look at this that it's fallen out of date. Can you resolve the conflicts and I can take another look?

**Comment by zaidalyafeai** [comment]
If u need some help I can review the prompts.

**Comment by KhalidAlt** [comment]
Hi @zaidalyafeai, It will be so helpful if you can review the Arabic prompts. Thanks a lot.

**Comment by KhalidAlt** [comment]
Hi @zaidalyafeai, I think you can review the prompts much easier now. I fixed the Unicode characters problem and it shows the Arabic characters now.

**Comment by KhalidAlt** [comment]
Hi @stephenbach, I fixed all the conflicts and a small bug that prevent Unicode characters from showing properly in .yaml files. Please let me know if there is anything need change. Also, show_new_templates check is always failing due to large size of data.

**Comment by zaidalyafeai** [comment]
@KhalidAlt the branch seems to include many other commits for other languages?

**Comment by KhalidAlt** [comment]
@zaidalyafeai, previously, Unicode characters were saved on byte format. Therefore, I fixed the issue, and most prompts have changed from byte to Unicode. This is the reason that there are many other commits for other languages. Please let me know if anything to be changed. Thanks

**Comment by zaidalyafeai** [comment]
Sure will review prompts today.

File: promptsource/templates/tydiqa/primary_task/templates.yaml [hunk 1]
```diff
@@ -353,48 +443,58 @@ templates:
...
       - Accuracy
       original_task: false
     name: jp_open_domain_qa
     reference: Ask Yes/No question
+  d9888ddb-c3e3-47a5-94ab-845057c98dac: !Template
+    answer_choices: نعم ||| ﻻ
+    id: d9888ddb-c3e3-47a5-94ab-845057c98dac
+    jinja: "{% if language == \"arabic\" %}\n {% if annotations.yes_no_answer[0] ==\
+      \ \"YES\" or annotations.yes_no_answer[0] == \"NO\" %} \nبعد قراءة النص المقتطع\
+      \ من ويكبيديا أجب عن السؤال التالي:\n{{question_text}} \n{{document_plaintext}}\n\
```
Comment (zaidalyafeai) [inline promptsource/templates/tydiqa/primary_task/templates.yaml:475]: المقتبس؟

File: promptsource/templates/tydiqa/secondary_task/templates.yaml [hunk 2]
```diff
@@ -383,36 +370,44 @@ templates:
...
       - BLEU
       - ROUGE
       original_task: false
     name: jp_end_to_end_question_generation
     reference: generate end-to-end question from a passage.
+  be8e455c-f3d2-434f-ab56-d1926252b691: !Template
+    answer_choices: null
+    id: be8e455c-f3d2-434f-ab56-d1926252b691
+    jinja: "{% set lang = id.split(\"-\")[0]%}     \n      {% if lang == \"arabic\"\
+      \ %}\nلطالما تسائلت حول: {{question}}\nبحث في ويكيبيديا ووجدت نص بعنوان {{title}}.\n\
```
Comment (zaidalyafeai) [inline promptsource/templates/tydiqa/secondary_task/templates.yaml:389]: بحثت

File: promptsource/templates/tydiqa/secondary_task/templates.yaml [hunk 3]
```diff
@@ -545,28 +536,70 @@ templates:
...
       metrics:
       - Squad
       original_task: true
     name: id_can_you_tell_me_the_answer
     reference: Ask question with Context
+  e8157b4f-17bc-42fc-9655-05a555822337: !Template
+    answer_choices: null
+    id: e8157b4f-17bc-42fc-9655-05a555822337
+    jinja: "{% set lang = id.split(\"-\")[0]%}     \n      {% if lang == \"arabic\"\
+      \ %}\nالنص: {{context}}\nاجابة هذا السؤال  \"{{question}}\" موجودة في النص المقتطع\
```
Comment (zaidalyafeai) [inline promptsource/templates/tydiqa/secondary_task/templates.yaml:550]: كلمة النص متكررة مرتان

File: promptsource/templates/tydiqa/secondary_task/templates.yaml [hunk 4]
```diff
@@ -545,28 +536,70 @@ templates:
...
+      \ بعنوان: {{title}}. هل يمكنك الإجابة عن السؤال؟\n|||\n{{answers.text | choice}}\n\
```
Comment (zaidalyafeai) [inline promptsource/templates/tydiqa/secondary_task/templates.yaml:551]: Is `answer.text` yes or no? The question seems to require that.

**Review by zaidalyafeai — CHANGES_REQUESTED** [review]

**Comment by samsontmr** [comment]
@KhalidAlt @stephenbach Any chance we can get this merged soon? We need the filter to be removed before we can run the evaluations for the paper. Thanks!

**Comment by VictorSanh** [comment]
@KhalidAlt it the tests pass locally, and @zaidalyafeai has validated, i can merge this in

**Comment by stephenbach** [comment]
Whoops, I DMed @KhalidAlt. Please don’t merge yet. There’s some code duplication in templates.py that needs to be removed.

**Comment by KhalidAlt** [comment]
Hi @stephenbach, I removed the code duplication. I used the argument (allow_unicode=True) because non-ASCII language prompts were saved in byte formatting which makes it difficult to review them manually using a .ymal file.

**Comment by stephenbach** [comment]
There's a .app.py.swp in the changes.

**Comment by KhalidAlt** [comment]
Hi @stephenbach, I have deleted it in the cleaning commit.

**Comment by stephenbach** [comment]
For the record, show_new_templates fails because the dataset is too big to run on GitHub and check_templates fails because of the issue that #832 addresses.

**Review by stephenbach — APPROVED** [review]
LGTM!

**Pull request merged**